internal/api/token.go

```diff
@@ -353,6 +353,11 @@ func (a *API) generateAccessToken(ctx context.Context, tx *storage.Connection, u
 			return "", 0, err
 		}
 		goTrueClaims := jwt.MapClaims(output.Claims)
+		appMetadata, ok := goTrueClaims["app_metadata"].(map[string]interface{})
```

Comment:
Can we update the entire claims instead of just `app_metadata`? E.g. someone might put the claim straight onto `event.claims` and then it wouldn't be reflected in the response, right?

Reply:
Yes that's right, a claim put straight onto `event.claims` wouldn't be reflected in the response.  

 The hope is that the requirement to add it on `app_metadata` would ensure it was intentionally added and that this would allow devs to unintentionally override other key claims (e.g. `email` / `phone`) which might lead to unexpected behaviour.  Restricting modification to `app_metadata` might make it easier to manage.

Let me check if there are any claims that would need to be on the top level and then circle back. Feel free to let me know if there's anything else I'm missing though

Reply:
Hm, since auth hooks is such an advanced feature I wouldn't see the need to add these guardrails and rather think that this would be unexpected to sync some parts of the response but not all. 

Should you decide to proceed with this approach, then however we should update the docs to make this clear, and also make sure the docs don't set claims directly on the user object as they currently do here for the user_level: https://supabase.com/docs/guides/auth/auth-hooks?language=add-metadata-claim-plv8#hook-custom-access-token

Reply:
@thorwebdev after discussion we settled on syncing the entire response - we'll add an advisory in the docs for devs to namespace their claims/fields and not use any fields which would be reserved by Supabase

Reply:
After further discussion,  we realized it's probably not possible to use a hook response of
```
{
   "myclaim1": "cl1",
   "myclaim2": "cl2",
   "myclaim3": "cl3",
   "app_metadata": {
      ...
    }
   "user_metadata" :{
        ...
    }
    ...
}
```
and return

```
{
  "token": "...",
  "refresh_token": "...",
  "user": {
     "user_metadata": {
      
     },
     "app_metadata": {
       
      },
      "myclaim1":"cl1",
      "myclaim2": "cl2",
      "myclaim3": "cl3",
   },
}
```

since the claims are [on the user model](https://github.com/supabase/gotrue/blob/master/internal/models/user.go#L32) and changing the response would mean changing the DB

Reply:
But wait, the point of the hook is that you don't need to be dependent on the DB. If you change `app_metadata` or `user_metadata` that's absolutely fine. It doesn't mean a change in the DB.